src/main/kotlin/FiToBookConvertor.kt

```diff
@@ -11,76 +11,112 @@ import java.time.format.DateTimeFormatter
 
 class FiToBookConvertor {
 
-    private val dateCol = "A"
-    private val shareCol = "B"
-    private val opCol = "C"
-    private val priceCol = "D"
-    private val feeCol = "E"
-    private val amountCol = "F"
-
-    /**
-     * Processes activities and converts them based on the specified mode
-     * @param activities List of activities to process
-     * @param mode Processing mode (0 or 1)
-     * @param startingRow Starting row number for processing
-     */
+    private companion object {
+        const val DATE_COL = "A"
+        const val SHARE_COL = "B"
+        const val OP_COL = "C"
+        const val PRICE_COL = "D"
+        const val FEE_COL = "E"
+        const val AMOUNT_COL = "F"
+        const val NEWLINE = '\n'
+        const val TAB = '\t'
+        val TRADE_TYPES = setOf(ActivityType.BOUGHT, ActivityType.SOLD)
+    }
+
+    private val dateFormatter = DateTimeFormatter.ofPattern("MM/dd/yyyy")
+
     fun processActivities(
         activities: List<Activity>,
         mode: String,
         startingRow: Int,
+        summRow: Int? = null,
+        summAmountCol: String? = null,
+        summBalCol: String? = null
+    ) {
+        val trades = activities.filter { it.type in TRADE_TYPES }
+        val toClipboard = buildString { // Inline string building: Used buildString for better memory allocation
+            when (mode) {
+                "0" -> processForExeTab(trades, startingRow)
+                "1" -> processForSummary(activities, startingRow, trades.size, summRow, summAmountCol, summBalCol)
+            }
+            // Remove trailing newline
+            if (isNotEmpty() && last() == NEWLINE) deleteCharAt(length - 1)
+        }
+
+        println("New clipboard:")
+        println(toClipboard)
+        writeToClipboard(toClipboard)
+        printDiscrepancies(trades)
+    }
+
+    private fun StringBuilder.processForExeTab(trades: List<Activity>, startingRow: Int) {
+        trades.forEachIndexed { index, trade ->
+            append(formatExecutionRow(trade, startingRow + index))
```

Comment:
```suggestion
            append(toInvestBookExecutionFromActivity(trade, startingRow + index))
```
keep original method name